skactiveml/pool/_bald.py

```diff
@@ -0,0 +1,228 @@
+        utilities : numpy.ndarray of shape (batch_size, n_samples) or
+                numpy.ndarray of shape (batch_size, n_candidates)
+            The utilities of samples after each selected sample of the batch,
+            e.g., `utilities[0]` indicates the utilities used for selecting
+            the first sample (with index `query_indices[0]`) of the batch.
+            Utilities for labeled samples will be set to np.nan.
+            If candidates is None or of shape (n_candidates), the indexing
+            refers to samples in X.
+            If candidates is of shape (n_candidates, n_features), the indexing
+            refers to samples in candidates.
+        """
+        # Validate input parameters.
+        X, y, candidates, batch_size, return_utilities = self._validate_data(
+            X, y, candidates, batch_size, return_utilities, reset=True
+        )
+
+        X_cand, mapping = self._transform_candidates(candidates, X, y)
+
+        # Validate classifier type.
+        check_type(fit_ensemble, "fit_ensemble", bool)
+
+        ensemble, est_arr, _ = _check_ensemble(
+            ensemble=ensemble,
+            X=X,
+            y=y,
+            sample_weight=sample_weight,
+            fit_ensemble=fit_ensemble,
+            missing_label=self.missing_label_,
+            estimator_types=[SkactivemlClassifier],
+        )
+
+        probas = np.array(
+                [est.predict_proba(X_cand) for est in est_arr]
+            )
+        batch_utilities_cand = _batchbald(probas, batch_size)
+
+        if mapping is None:
+            batch_utilities = batch_utilities_cand
+        else:
+            batch_utilities = np.full((batch_size, len(X)), np.nan)
+            batch_utilities[:, mapping] = batch_utilities_cand
+
+        best_indices = rand_argmax(batch_utilities,
+                                   axis=1,
+                                   random_state=self.random_state_)
+
+        if return_utilities:
+            return best_indices, batch_utilities
+        else:
+            return best_indices
+
+
+def _bald(probas):
+    """Bayesian Active Learning by Disagreement (BALD)
+    Computes the Bayesian Active Learning by Disagreement (BALD) score for
+    each sample.
+
+    Parameters
+    ----------
+    probas : array-like, shape (n_estimators, n_samples, n_classes)
+        The probability estimates of all estimators, samples, and classes.
+
+    Returns
+    -------
+    scores: np.ndarray, shape (n_samples)
+        The BALD-scores.
+
+    References
+    ----------
+    [1] Houlsby, Neil, et al. "Bayesian active learning for classification and
+        preference learning." arXiv preprint arXiv:1112.5745 (2011).
+    """
+    p_mean = np.mean(probas, axis=0)
+    uncertainty = np.nansum(-p_mean * np.log(p_mean), axis=1)
+    confident = np.nanmean(np.nansum(-probas * np.log(probas), axis=2), axis=0)
+    return uncertainty - confident
+
+
+def _batchbald(probas, batch_size=1):
```

Comment:
```suggestion
def batch_bald(probas, batch_size=1):
```
Add check for `probas` and `batch_size`